src/main/java/application/MenuController.java

```diff
@@ -249,4 +260,10 @@ private MenuSession performInstall(InstallRequestBean bean, String authToken) th
                 bean.getInstallReference(), bean.getLocale(), installService, restoreFactory, auth, host,
                         bean.getOneQuestionPerScreen(), bean.getRestoreAs());
     }
+
+    private MenuSession performUpdate(UpdateRequestBean updateRequestBean, String authToken) throws Exception {
+        MenuSession currentSession = performInstall(updateRequestBean, authToken);
+        currentSession.updateApp(updateRequestBean.getUpdateMode());
```

Comment:
how does this exactly work? given an update mode it will query HQ at a certain URL to ensure that it gets the latest release/build/save for that app id?